### PR TITLE
DT-1058 - making IAO and IAM fields mandatory

### PR DESCRIPTION
**Pull request by tayyib-saddique**
### Description of change

Makes IAO and IAM fields for all catalogue items (with the exception of Reference Datasets (separate ticket)

Screenshot 2023-02-14 at 11 32 24

Error message if field is left blank

Screenshot 2023-02-14 at 11 32 50

### Checklist

* [ ] Have tests been added to cover any changes?

### CODE REVIEW

File: dataworkspace/dataworkspace/apps/dw_admin/forms.py [hunk 1]
```diff
@@ -49,7 +49,6 @@ def __init__(self, *args, **kwargs):
 
 
 class ReferenceDatasetForm(AutoCompleteUserFieldsMixin, forms.ModelForm):
-    model = ReferenceDataset
```
Comment (nboyse) [inline dataworkspace/dataworkspace/apps/dw_admin/forms.py:52]: Ah - we shouldn't delete this or the form will break

File: dataworkspace/dataworkspace/apps/dw_admin/forms.py [hunk 2]
```diff
@@ -58,6 +57,12 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if "sort_field" in self.fields:
             self.fields["sort_field"].queryset = self.instance.fields.all()
+        # self.fields["information_asset_owner"].required = True
```
Comment (nboyse) [inline dataworkspace/dataworkspace/apps/dw_admin/forms.py:60]: I would remove these changes for now (commented lines and Meta class)

File: dataworkspace/dataworkspace/static/js/react_apps/package-lock.json [hunk 3]
```diff
@@ -1196,9 +1196,9 @@
       }
     },
     "@babel/runtime": {
-      "version": "7.20.7",
-      "resolved": "https://registry.npmjs.org/@babel/runtime/-/runtime-7.20.7.tgz",
-      "integrity": "sha512-UF0tvkUtxwAgZ5W/KrkHf0Rn0fdnLDU9ScxBrEVNUprE/MzirjK4MJUX1/BVDv00Sv8cljtukVK1aky++X1SjQ==",
+      "version": "7.20.13",
```
Comment (nboyse) [inline dataworkspace/dataworkspace/static/js/react_apps/package-lock.json:1199]: I don't think any front end changes/compiles should be in this PR

File: dataworkspace/dataworkspace/tests/factories.py [hunk 4]
```diff
@@ -175,6 +177,8 @@ class ReferenceDatasetFactory(factory.django.DjangoModelFactory):
     published = True
     schema_version = factory.Sequence(lambda n: n)
     table_name = factory.fuzzy.FuzzyText(length=20, prefix="ref_")
+    information_asset_owner = factory.SubFactory(UserFactory)
+    information_asset_manager = factory.SubFactory(UserFactory)
```
Comment (nboyse) [inline dataworkspace/dataworkspace/tests/factories.py:181]: If we are doing reference separately, shouldn't we leave these changes out for now?

File: dataworkspace/dataworkspace/tests/test_admin.py [hunk 5]
```diff
@@ -454,6 +454,8 @@ def test_create_reference_dataset_existing_table_name(self):
                 "valid_from": "",
                 "valid_to": "",
                 "enquiries_contact": "",
+                "information_asset_owner": self.user.id,
+                "information_asset_manager": self.user.id,
```
Comment (nboyse) [inline dataworkspace/dataworkspace/tests/test_admin.py:458]: Same with these reference table tests, what do you think?

File: dataworkspace/dataworkspace/tests/test_admin.py [hunk 6]
```diff
@@ -3507,6 +3572,8 @@ def test_source_table_reporting_disabled(self, staff_client):
                 "slug": dataset.slug,
                 "short_description": "test short description",
                 "description": "test description",
+                "information_asset_owner": user.id,
+                "information_asset_manager": user.id,
```
Comment (nboyse) [inline dataworkspace/dataworkspace/tests/test_admin.py:3576]: A few inconsistencies between str() and leaving it in id form, shall we convert all to str?

File: requirements.txt [hunk 7]
```diff
@@ -79,7 +79,7 @@ cryptography==38.0.3
     # via -r requirements.in
 dj-database-url==0.5.0
     # via -r requirements.in
-django==3.2.16
+django==3.2.17
```
Comment (nboyse) [inline requirements.txt:82]: Ah, should this django upgrade be in a separate PR?

**Review by nboyse — CHANGES_REQUESTED** [review]

File: dataworkspace/dataworkspace/static/js/react_apps/package-lock.json (hunk 3 above)
Comment (tayyib-saddique) [reply dataworkspace/dataworkspace/static/js/react_apps/package-lock.json:1199]: believe this is due to merge

File: requirements.txt (hunk 7 above)
Comment (tayyib-saddique) [reply requirements.txt:82]: due to merge earlier I believe

File: dataworkspace/dataworkspace/tests/test_admin.py (hunk 6 above)
Comment (tayyib-saddique) [reply dataworkspace/dataworkspace/tests/test_admin.py:3576]: have removed all instances of str() form (they all relate to referencedataset rather than datasets) 
str() form was added as the user.id came from `UserFactory` rather than `get_user_model() `

**Review by nboyse — APPROVED** [review]
Fab work
